### PR TITLE
Make sure CoberturaXMLParser is not trying to access external network…

### PR DESCRIPTION
**Pull request by artms**
Xerces is actually trying to download document schema from internet, eg.: `http://cobertusa.sourceforge.net/xml/coverage-03.dtd` and from time to time it fails and this results in failed coverage data upload to phabricator.

We "statically" downloaded DTD file and return them back if parser request, if it requests something unknown then we return empty document

### CODE REVIEW

File: src/main/java/com/uber/jenkins/phabricator/coverage/CoberturaXMLParser.java [hunk 1]
```diff
@@ -47,9 +52,20 @@
     private static final String NODE_NAME_LINE = "line";
     private static final String NODE_NUMBER = "number";
     private static final String NODE_HITS = "hits";
+    private static final String EMPTY_XML = "<?xml version='1.0' encoding='UTF-8'?>";
+    private static final Logger LOGGER = Logger.getLogger(CoberturaXMLParser.class.getName());
+    private static final Map<String, String> dtdMap = new HashMap<String, String>();
```
Comment (kageiit) [inline src/main/java/com/uber/jenkins/phabricator/coverage/CoberturaXMLParser.java]: Use an immutableList from guava and built this in place

File: src/main/java/com/uber/jenkins/phabricator/coverage/CoberturaXMLParser.java [hunk 2]
```diff
@@ -47,9 +52,20 @@
...
+
     private final FilePath workspace;
     private final Set<String> includeFileNames;
 
+    static {
```
Comment (kageiit) [inline src/main/java/com/uber/jenkins/phabricator/coverage/CoberturaXMLParser.java]: Avoid static blocks. build list at the time you declare it

File: src/main/java/com/uber/jenkins/phabricator/coverage/CoberturaXMLParser.java [hunk 3]
```diff
@@ -66,6 +82,18 @@
             try {
                 is = new FileInputStream(file);
                 db = dbf.newDocumentBuilder();
+                db.setEntityResolver(new EntityResolver() {
```
Comment (kageiit) [inline src/main/java/com/uber/jenkins/phabricator/coverage/CoberturaXMLParser.java]: Remove anonymous inner class.

Make the EntityResolver a private static final constant that is initialized at the beginning

**Comment by artms** [comment]
@kageiit - comments were addressed

**Pull request merged**